dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+
+                    Logger.error(this, e.getMessage(), e);
+                    failedInodes.add(templateInode);
+                }
+            } else {
+
+                failedInodes.add(templateInode);
+            }
+        }
+
+        return Response.ok(new ResponseEntityView(
+                CollectionsUtils.map(
+                "publishedInodes", publishedInodes,
+                "failedInodes",   failedInodes
+                ))).build();
+    }
+
+    /**
+     * Unpublish a list of template inodes
+     * Return the list of a success unpublished and failed
+     * @param request            {@link HttpServletRequest}
+     * @param response           {@link HttpServletResponse}
+     * @param templatesToUnpublish {@link List} list of template inodes to unpublish
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @PUT
+    @Path("/_unpublish")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response unpublish(@Context final HttpServletRequest  request,
+                                  @Context final HttpServletResponse response,
+                                  final List<String> templatesToUnpublish) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user         = initData.getUser();
+        final Host host         = this.hostWebAPI.getCurrentHostNoThrow(request);
+        final PageMode pageMode = PageMode.get(request);
+        final List<String> publishedInodes = new ArrayList<>();
+        final List<String> failedInodes    = new ArrayList<>();
+
+        if (!UtilMethods.isSet(templatesToUnpublish)) {
+
+            throw new IllegalArgumentException("The body must send a collection of template inodes such as: " +
+                    "[\"dd60695c-9e0f-4a2e-9fd8-ce2a4ac5c27d\",\"cc59390c-9a0f-4e7a-9fd8-ca7e4ec0c77d\"]");
+        }
+
+        for (final String templateInode : templatesToUnpublish) {
+
+            final Template template = this.templateAPI.find(templateInode, user, pageMode.respectAnonPerms);
+
+            if (null != template && InodeUtils.isSet(template.getInode())) {
+
+                try {
+
+                    if (this.templateAPI.unpublishTemplate(template, user, pageMode.respectAnonPerms)) {
+
+                        ActivityLogger.logInfo(this.getClass(), "UnPublish Template action", "User " +
+                                user.getPrimaryKey() + " unpublishing template" + template.getTitle(), host.getTitle() != null ? host.getTitle() : "default");
+                        publishedInodes.add(templateInode);
+                    } else {
+                        failedInodes.add(templateInode);
+                    }
+                } catch(Exception wax) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)